### PR TITLE
Adding user Associations APIs

### PR DESCRIPTION
**Pull request by IsuraD**

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/AssociationEndpointConstants.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.identity.rest.api.user.association.v1;
+
+public class AssociationEndpointConstants {
+    public static final String HEADER_CONTENT_TYPE = "Content-Type";
+    public static final String APPLICATION_JSON = "application/json";
+    public static final String DEFAULT_RESPONSE_CONTENT_TYPE = APPLICATION_JSON;
+    public static final String STATUS_BAD_REQUEST_MESSAGE_DEFAULT = "Bad Request";
```
Comment (ayshsandu) [inline components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/AssociationEndpointConstants.java]: retrieve from Response.Status.BAD_REQUEST instead defining constants

File: components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/AssociationEndpointConstants.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.identity.rest.api.user.association.v1;
+
+public class AssociationEndpointConstants {
+    public static final String HEADER_CONTENT_TYPE = "Content-Type";
+    public static final String APPLICATION_JSON = "application/json";
+    public static final String DEFAULT_RESPONSE_CONTENT_TYPE = APPLICATION_JSON;
```
Comment (ayshsandu) [inline components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/AssociationEndpointConstants.java]: Use HttpHeaders.CONTENT_TYPE and MediaType.APPLICATION_JSON

File: components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/impl/UserIdApiServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+public class UserIdApiServiceImpl extends UserIdApiService {
+
+    @Autowired
+    private UserAssociationService userAssociationService;
+
+    @Override
+    public Response userIdAssociationsDelete(String userId) {
+
+        userAssociationService.deleteUserAccountAssociation(userId);
```
Comment (ayshsandu) [inline components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/impl/UserIdApiServiceImpl.java]: userId= base64encode(Userstore/username) format. Hence, decode

File: components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/resources/META-INF/cxf/user-association-v1-cxf.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<beans xmlns="http://www.springframework.org/schema/beans" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:jaxrs="http://cxf.apache.org/jaxrs" xmlns:context="http://www.springframework.org/schema/context" xsi:schemaLocation=" http://www.springframework.org/schema/beans  http://www.springframework.org/schema/beans/spring-beans-3.0.xsd http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context-3.0.xsd http://cxf.apache.org/jaxrs http://cxf.apache.org/schemas/jaxrs.xsd">
+    <bean class="org.wso2.carbon.identity.rest.api.user.association.v1.core.UserAssociationService"/>
+    <bean class="org.wso2.carbon.identity.rest.api.user.association.v1.impl.UserIdApiServiceImpl"/>
+    <bean class="org.wso2.carbon.identity.rest.api.user.association.v1.impl.MeApiServiceImpl"/>
+</beans>
```
Comment (ayshsandu) [inline components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/resources/META-INF/cxf/user-association-v1-cxf.xml]: newline at the end

File: components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/core/UserAssociationService.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        userDTO.setUsername(userAccountAssociationDTO.getUsername());
+        userDTO.setUserStoreDomain(userAccountAssociationDTO.getDomain());
+        return userDTO;
+    }
+
+    private APIError handleUserAccountAssociationException(UserAccountAssociationException e) {
+
+        ErrorResponse errorResponse = new ErrorResponse.Builder()
+                .withCode(e.getErrorCode())
+                .withMessage(e.getMessage())
```
Comment (ayshsandu) [inline components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/core/UserAssociationService.java]: send short message

File: components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/impl/MeApiServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    private String getUserId() {
+
+        String username = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        return UserCoreUtil.addTenantDomainToEntry(username, PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                .getTenantDomain());
+    }
+
+    private URI getAssociationsLocationURI(String userId) throws URISyntaxException {
+
+        return new URI("/" + userId + "/" + ASSOCIATIONS_PATH);
```
Comment (ayshsandu) [inline components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/impl/MeApiServiceImpl.java]: use the util in https://github.com/wso2/identity-api-user/blob/master/components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/ContextLoader.java#L109 to build fully qualified path

File: components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/AssociationEndpointConstants.java (hunk 1 above)
Comment (IsuraD) [reply components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/AssociationEndpointConstants.java]: done

File: components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/impl/UserIdApiServiceImpl.java (hunk 3 above)
Comment (IsuraD) [reply components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/impl/UserIdApiServiceImpl.java]: will do this later.

File: components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/impl/MeApiServiceImpl.java (hunk 6 above)
Comment (IsuraD) [reply components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/impl/MeApiServiceImpl.java]: done

**Pull request merged**